miio/integrations/vacuum/roborock/vacuumcontainers.py

```diff
@@ -178,18 +224,49 @@ def battery(self) -> int:
         step=1,
         icon="mdi:fan",
     )
-    def fanspeed(self) -> int:
+    def fanspeed(self) -> Optional[int]:
         """Current fan speed."""
-        return int(self.data["fan_power"])
+        fan_power = int(self.data["fan_power"])
+        if fan_power > 100:
+            # values 100+ are reserved for presets
+            return None
+        return fan_power
 
     @property
-    @sensor("Clean Duration", unit="s", icon="mdi:timer-sand")
+    @setting(
+        "Mop scrub intensity",
+        choices=MopIntensity,
+        setter_name="set_mop_intensity",
+        icon="mdi:checkbox-multiple-blank-circle-outline",
+    )
+    def mop_intensity(self) -> int:
+        """Current mop intensity."""
+        return int(self.data["water_box_mode"])
```

Comment:
This needs to be changed to be optional, too.

Reply:
done